src/pages/Dashboard.jsx

```diff
@@ -17,72 +19,91 @@ const LAST_THREE_GROUPS = MOCK_GROUP_INFO.slice(-3);
-                    <CardHeader>
-                        <UsersRound />
+                <Card className="w-full mt-4 md:m-0">
+                    <CardHeader className="flex flex-row my-auto">
+                        <UsersRound size={64} className="pr-3 border-r-2" />
+                        <div className="flex flex-col ml-4">
+                            <Heading tag="h2" className="mb-0 md:mb-0 lg:text-2xl">Total Friends</Heading>
+                            <Heading tag="h2" className="mb-0 lg:text-2xl">2999</Heading>
+                        </div>
                     </CardHeader>
                 </Card>
-                <Card className="w-full mt-4">
-                    <CardHeader>
-                        <CircleDollarSign />
+                <Card className="w-full mt-4 md:m-0">
+                    <CardHeader className="flex flex-row my-auto">
+                        <CircleDollarSign size={64} className="pr-3 border-r-2" />
+                        <div className="flex flex-col ml-4">
+                            <Heading tag="h2" className="mb-0 md:mb-0 lg:text-2xl">Total Expenses</Heading>
+                            <Heading tag="h2" className="mb-0 lg:text-2xl">{totalExpenses(MOCK_EXPENSES)}</Heading>
+                        </div>
                     </CardHeader>
                 </Card>
             </section>
+            <section className="w-full">
+                <Heading tag="h1" className="mt-10 mb-5">Group Expense Overview</Heading>
+                <div className="w-full md:flex md:gap-2">
+                    {
+                        LAST_THREE_GROUPS.map((group) => {
+                            return (
+                                <Card key={group.id} className="w-1/3">
+                                    <CardHeader className="flex flex-row items-center max-w-max">
+                                        <Avatar className="size-16">
+                                            <AvatarImage src={group.avatar} />
+                                            <AvatarFallback className="bg-slate-300 text-foreground">
+                                                {getInitials(group.name)}
+                                            </AvatarFallback>
+                                        </Avatar>
+                                        <Heading tag="h2" className="ml-3">{group.name}</Heading>
+                                    </CardHeader>
+                                    <CardContent className="ml-5">
+                                        <BodyText tag="p" variant='normal' className="mb-1">Alloted Budget: ${group.budget}</BodyText>
+                                        <BodyText tag="p" variant='normal' className="mb-1">Expense Amount: ${group.expense}</BodyText>
+                                        <BodyText tag="p" variant='normal' className="mb-1">Participants: ${group.members}</BodyText>
+                                    </CardContent>
```

Comment:
Please consider using `Button` component with the `primary` text color and the `variant='link'` and `asChild` prop for the to improve visual consistency. Example:

```jsx
<Button className="text-primary" variant="link" asChild>
    <NavLink to={`/app/groups/${group.id}`}>See More</NavLink>
</Button>
```